core/scripts.py

```diff
@@ -2804,46 +2798,37 @@ def do_run(self, cmd_args: list[str] | None = None) -> None:
 
     def process_task(self, task: DeferredTask):
         data = InventoryReportTaskData(**task.data)
-        files = []
+        files: list[IO] = []
         try:
             current_time = datetime.datetime.now()
             date_str = current_time.strftime("%Y-%m-%d_%H:%M:%s")
-            attachments = {}
-
-            integrations = self._db.scalars(
-                select(IntegrationConfiguration)
-                .join(IntegrationLibraryConfiguration)
-                .where(
-                    IntegrationLibraryConfiguration.library_id == data.library_id,
-                    IntegrationConfiguration.goal == Goals.LICENSE_GOAL,
-                    not_(
-                        IntegrationConfiguration.settings_dict.contains(
-                            {"include_in_inventory_report": False}
-                        )
-                    ),
-                )
-            ).all()
+            attachments: dict[str, str] = {}
+            library = get_one(self._db, Library, id=data.library_id)
+            assert library
```

Comment:
If a library is deleted, between when the task is queued, and when its run, library will be `None`. Which will trigger this assertion. Maybe we want to handle this more gracefully then just dying with an `AssertionError`?

Reply:
I'll fix.